subsys/bluetooth/mesh/brg_cfg.c

```diff
@@ -0,0 +1,317 @@
+	} else {
+		err = settings_delete(path);
+	}
+
+	if (err) {
+		LOG_ERR("Failed to store %s value", path_bt);
+	}
+#endif
+}
+
+/* Remove the entry from the bridging table that corresponds with the netkey index
+ * of the removed subnet.
+ */
+static void brg_tbl_netkey_removed_evt(struct bt_mesh_subnet *sub, enum bt_mesh_key_evt evt)
+{
+	for (int i = 0; i < CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX; i++) {
+		if (brg_tbl[i].direction && (
+		    brg_tbl[i].net_idx1 == sub->net_idx ||
+		    brg_tbl[i].net_idx2 == sub->net_idx)) {
+			memset(&brg_tbl[i], 0, sizeof(brg_tbl[i]));
+			brg_tbl_compact();
+		}
+	}
+
+#if IS_ENABLED(CONFIG_BT_SETTINGS)
+	bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_BRG_PENDING);
+#endif
+}
+
+/* Add event hook for key deletion event */
+BT_MESH_SUBNET_CB_DEFINE(sbr) = {
+	.evt_handler = brg_tbl_netkey_removed_evt,
+};
+
+int brg_cfg_init(void)
+{
+	memset(brg_tbl, 0, sizeof(brg_tbl));
+	brg_tbl_row_cnt = 0;
+	return 0;
+}
+
+int brg_cfg_tbl_reset(void)
+{
+	brg_cfg_init();
+
+	int err = settings_delete("bt/mesh/brgen");
+
+	if (err) {
+		return err;
+	}
+	err = settings_delete("bt/mesh/brgbt");
+
+	return err;
+}
+
+int brg_cfg_tbl_get(const struct brg_tbl_row **rows)
+{
+	*rows = brg_tbl;
+	return brg_tbl_row_cnt;
+}
+
+int brg_cfg_tbl_add(enum brg_tbl_direction direction, uint16_t net_idx1, uint16_t net_idx2,
+			uint16_t addr1, uint16_t addr2)
+{
+	/* Sanity check */
+	if (!BT_MESH_ADDR_IS_UNICAST(addr1) || net_idx1 == net_idx2 || addr1 == addr2 ||
+	    net_idx1 > 0x03FF || net_idx2 > 0x03FF ||
+	    (direction != BRG_ADDR1_TO_ADDR2 && direction != BRG_ADDR1_TF_ADDR2) ||
+	    (direction == BRG_ADDR1_TO_ADDR2 &&
+		(addr2 == BT_MESH_ADDR_UNASSIGNED || addr2 == BT_MESH_ADDR_ALL_NODES)) ||
+	    (direction == BRG_ADDR1_TF_ADDR2 && !BT_MESH_ADDR_IS_UNICAST(addr2)))	{
+		return -EINVAL;
+	}
+
+	/* Check if entry already exists, if yes, then it is success. */
+	for (int i = 0; i < brg_tbl_row_cnt; i++) {
+		if (brg_tbl[i].direction == direction && brg_tbl[i].net_idx1 == net_idx1 &&
+		    brg_tbl[i].net_idx2 == net_idx2 && brg_tbl[i].addr1 == addr1 &&
+		    brg_tbl[i].addr2 == addr2) {
+			return 0;
+		}
+	}
+
+	/* Empty element, is the current table row counter */
+	if (brg_tbl_row_cnt == CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX) {
+		return -ENOMEM;
+	}
+
+	/* Update the row */
+	brg_tbl[brg_tbl_row_cnt].direction = direction;
+	brg_tbl[brg_tbl_row_cnt].net_idx1 = net_idx1;
+	brg_tbl[brg_tbl_row_cnt].net_idx2 = net_idx2;
+	brg_tbl[brg_tbl_row_cnt].addr1 = addr1;
+	brg_tbl[brg_tbl_row_cnt].addr2 = addr2;
+	brg_tbl_row_cnt++;
+
+#if IS_ENABLED(CONFIG_BT_SETTINGS)
+	bt_mesh_settings_store_schedule(BT_MESH_SETTINGS_BRG_PENDING);
+#endif
+
+	return 0;
+}
+
+static int itr;
+
+int brg_cfg_tbl_check_start(void)
+{
+	itr = 0;
+
+	if (!brg_tbl_row_cnt) {
+		return -ENOENT;
+	}
+
+	return 0;
+}
+
+int brg_cfg_tbl_check_itr(uint16_t src, uint16_t dst, uint16_t netidx, uint16_t *new_netidx)
```

Comment:
You do not need `itr` variable. It is sufficient to return\pass a bridge table entry pointer to continue iteration.

`brg_cfg_tbl_check_start` is not required. If pointer is passed into iteration function then passing NULL means start from the beginning.

Reply:
I don't think this works well for caller of the function in network layer. It requires unnecessary holding of the table pointer somehwere else, that is extra burden on the caller.

Reply:
There are already a couple of subnetworks iterator templates. https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/subnet.h#L110-L138
I pointed this: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/subnet.h#L129-L138
I'm pretty sure it is possible to reuse the same pattern here. 
No reason to create one more.